### PR TITLE
DR2-1655 Enforce queue and DLQ alarms.

### PR DESCRIPTION
**Pull request by MancunianSam**
I've removed the variable that turns off the dlq alarms and added in an
alarm for the queue itself as well. There are two new variables to
configure the thresholds.

### CODE REVIEW

File: sqs/main.tf [hunk 1]
```diff
@@ -80,3 +79,19 @@ module "dlq_cloudwatch_alarm" {
...
+  name                = "${var.queue_name}-alarm"
+  threshold           = var.queue_cloudwatch_alarm_visible_messages_threshold
+  comparison_operator = "GreaterThanThreshold"
+  statistic           = "Sum"
+  treat_missing_data  = "ignore"
+  datapoints_to_alarm = 1
+  dimensions = {
+    QueueName = local.sqs_queue.name
+  }
+  notification_topic = var.dlq_notification_topic
```
Comment (techncl) [inline sqs/main.tf:96]: Is `var.dlq_notification_topic` correct?

**Review by techncl — COMMENTED** [review]
Looks good! Just one question

File: sqs/main.tf [hunk 2]
```diff
@@ -65,12 +65,11 @@ resource "aws_sqs_queue" "dlq_with_sse" {
 }
 
 module "dlq_cloudwatch_alarm" {
-  count               = var.create_cloudwatch_alarm ? 1 : 0
   source              = "../cloudwatch_alarms"
   metric_name         = "ApproximateNumberOfMessagesVisible"
   namespace           = "AWS/SQS"
   name                = "${var.queue_name}-dlq-alarm"
```
Comment (kurtismash) [inline sqs/main.tf:71]: Worth adding something like "messages-visible" as the
 DLQ could have other alarms?

File: sqs/main.tf (hunk 1 above)
Comment (MancunianSam) [reply sqs/main.tf:96]: Ah no, it needs a new variable. I've added on in.

File: sqs/main.tf (hunk 2 above)
Comment (MancunianSam) [reply sqs/main.tf:71]: Good point, I've added it in.

Comment (kurtismash) [reply sqs/main.tf:71]: I think the DLQ will also need this.

Comment (MancunianSam) [reply sqs/main.tf]: Fair enough, I've changed this one as well.

**Review by kurtismash — APPROVED** [review]

**Pull request merged**